modin/backends/pandas/query_compiler.py

```diff
@@ -509,6 +509,83 @@ def is_monotonic_decreasing(self):
         ),
     )
 
+    def value_counts(self, **kwargs):
+        """
+        Return a QueryCompiler of Series containing counts of unique values.
+
+        Returns
+        -------
+        PandasQueryCompiler
+        """
+        if kwargs.get("bins", None) is not None:
```

Comment:
Do bins not work with `MapReduce` style? How is the designator for the bin determined?

Reply:
To see how is the designator for the bin determined it is need to see an example.
```python
import pandas
import numpy as np
s = pandas.Series([3, 1, 2, 3, 4, np.nan])
r = s.value_counts(bins=3)
r
(2.0, 3.0]      2
(0.996, 2.0]    2
(3.0, 4.0]      1
dtype: int64
r.index
IntervalIndex([(2.0, 3.0], (0.996, 2.0], (3.0, 4.0]],
              closed='right',
              dtype='interval[float64]')
```
`MapReduce` style doesn't work for bins because equal values may be put in to different bins at the `map` stage. Then, when we do `groupby` by index, we will not able to group indices with equal values.